### PR TITLE
[MLIR][NVVM] Enable import of nvvm.barrier0

### PR DESCRIPTION
**Pull request by ivanradanov**

### CODE REVIEW

**Review by wsmoses — APPROVED** [review]

File: mlir/include/mlir/Dialect/LLVMIR/NVVMOps.td [hunk 1]
```diff
@@ -431,7 +430,7 @@ def NVVM_MBarrierTestWaitSharedOp : NVVM_Op<"mbarrier.test.wait.shared">,
 // NVVM synchronization op definitions
 //===----------------------------------------------------------------------===//
 
-def NVVM_Barrier0Op : NVVM_Op<"barrier0"> {
+def NVVM_Barrier0Op : NVVM_IntrOp<"barrier0"> {
   string llvmBuilder = [{
```
Comment (schwarzschild-radius) [inline mlir/include/mlir/Dialect/LLVMIR/NVVMOps.td:434]: Given the `barrier0` is an NVVM_IntrOp, Can the llvmBuilder code be removed?

File: mlir/test/Target/LLVMIR/Import/nvvmir.ll [hunk 2]
```diff
@@ -71,12 +71,13 @@ define float @nvvm_rcp(float %0) {
   ret float %2
 }
 
-; TODO: Support the intrinsics below once they derive from NVVM_IntrOp rather than from NVVM_Op.
+define void @llvm_nvvm_barrier0() {
+  call void @llvm.nvvm.barrier0()
+  ret void
+}
```
Comment (gysit) [inline mlir/test/Target/LLVMIR/Import/nvvmir.ll:77]:
```suggestion
; CHECK-LABEL: @llvm_nvvm_barrier0()
define void @llvm_nvvm_barrier0() {
  ; CHECK: nvvm.barrier0
  call void @llvm.nvvm.barrier0()
  ret void
}
```
Can you add check lines to verify the right operation is produced.

**Review by gysit — APPROVED** [review]
Thanks, LGTM modulo comment.

**Comment by grypp** [comment]
The PR looks good, but we might want to deprecate this operation (`nvvm.barrier0`) and use   `nvvm.barrier` OP. It is more capable and aligns better with the PTX instruction.

**Comment by ivanradanov** [comment]
Thank you for the comments.

**Pull request merged**